spec/controllers/cases/offender_sar_controller_spec.rb

```diff
@@ -68,6 +68,159 @@
     end
   end
 
+  describe '#create' do
+    before do
+      sign_in responder
+      post :create, params: params
+      expect(assigns(:case)).to be_a Case::SAR::Offender
+      expect(flash[:notice]).to eq nil
+    end
+
+    context 'partial validations' do
+      context 'for step subject-details' do
+        let(:params) do
+          {
+            current_step: 'subject-details',
+            offender_sar: {
+              subject_full_name: '',
+              subject_address: '',
+            }
+          }
+        end
+
+        it 'validates subject name and address' do
+          errors = assigns(:case).errors.messages
+          expect(errors[:subject_full_name]).to eq ["can't be blank"]
+          expect(errors[:subject_address]).to eq ["can't be blank"]
+        end
+
+        it 'sets empty values and validates' do
+          errors = assigns(:case).errors.messages
+          expect(errors[:date_of_birth]).to eq ["can't be blank"]
+          expect(errors[:subject_type]).to eq ["can't be blank"]
+          expect(errors[:flag_as_high_profile]).to eq ["can't be blank"]
+        end
+      end
+
+      context 'for step requester-details' do
+        context 'when third party absent' do
+          let(:params) do
+            {
+              current_step: 'requester-details',
+              offender_sar: {
+                third_party_relationship: '',
+                third_party_name: '',
+                third_party_company_name: '',
+                postal_address: '',
+              }
+            }
+          end
+
+          it 'requires third_party to be set' do
+            errors = assigns(:case).errors.messages
+            expect(errors[:third_party]).to eq ["can't be blank"]
+          end
+        end
+
+        context 'when third party true' do
+          let(:params) do
+            {
+              current_step: 'requester-details',
+              offender_sar: {
+                third_party: true,
+                third_party_relationship: '',
+                third_party_name: '',
+                third_party_company_name: '',
+                postal_address: '',
+              }
+            }
+          end
+
+          it 'validates requester details' do
+            errors = assigns(:case).errors.messages
+            expect(errors[:third_party_name]).to eq ["can't be blank if company name not given"]
```

Comment:
Not a big thing but we could wrap these expects in a method as they are repeated on lines `208-211`. Method could be something like:

`expect_third_party_details_to_be_correct`

Reply:
Done, thanks!